### DISCUSSION TITLE
ST_Read for .osm.pbf mises some closed ways to convert to polygon

### DISCUSSION THREAD

**Discussion by EvgeniyPaskin** [duckdb/duckdb-spatial#424] [Q&A]
I'm trying to read .osm.pbf data to DuckDB for further analysis with help of truly wonderful _spatial_ extension. 
I've spotted that at least some of OSM ways are not converted to polygons for some reason and I can't really understand why.

Reading data with ST_ReadOSM
 `LOAD spatial;
  SELECT * FROM ST_ReadOSM('central_fed_district.osm.pbf')
  WHERE id = '60434574'`

shows the required object:
`way,60434574,"{amenity=place_of_worship, name=Церковь Рождества Пресвятой Богородицы, disused=yes, ref:temples.ru=6946, building=church, denomination=russian_orthodox, religion=christian, start_date=1798..1803}","{752330226,752330238,752330228,752330233,752330226}",,,,
`
it is of rectangular shape and as can be seen it is "closed"

When I use ST_Read:
```
SELECT *
FROM ST_READ('central_fed_district.osm.pbf',
                          open_options=['INTERLEAVED_READING=YES', 'CONFIG_FILE=osmconf.ini']
                          --layer='multipolygons',
                          --sequential_layer_scan=false  --tried true as well
                          )
WHERE osm_way_id = '60434574'
```
My understanding was that using GDAL's osmconf.ini with
`closed_ways_are_polygons=amenity,aeroway,waterway,boundary,building,craft,geological,historic,landuse,leisure,military,natural,office,place,shop,sport,tourism,highway=platform,public_transport=platform
`
should force conversion of the amenity object to polygon.

BUT, nothing shows up as a result of ST_Read query (I also tried to look manually for all objects within range of the target object with no luck).

Would appreciate any help/direction where to dig.

**Comment by Maxxen** [comment]
Not sure, but you don't pass named parameters to table functions with `--opt=arg`, but simply `opt=arg`
